src/move/move.ts

```diff
@@ -0,0 +1,23 @@
+const ALLOWED_FIRST_MOVES = [
+  [false, true, true, false],
+  [true, false, false, false, true],
+  [true, false, false, false, false, true],
+  [false, false, false, false, false, false, false],
+  [true, false, false, false, false, true],
+  [true, false, false, false, true],
+  [false, true, true, false],
+];
+
+export interface Action {
+  x: number;
+  y: number;
+  isFirstMove: boolean;
+}
```

Comment:
why not call this type `Move` to be more consistent with other variable and fn names?

Also, you will probably need to manipulate only coordinates at some point, so you might as well create a Coordinates type:

```ts
export interface Coordinates {
  x: number;
  y: number;
}

export interface Action {
  coordinates: Coordinates;
  isFirstMove: boolean;
}
```

Reply:
I just kept coordinates and moved whole "actions" concept to gameState